astropy/utils/tests/test_data.py

```diff
@@ -234,6 +234,43 @@ def a_file(tmp_path):
     yield fn, contents
 
 
+def test_temp_cache(tmpdir):
+    dldir0, urlmapfn0 = _get_download_cache_locs()
+    check_download_cache(check_hashes=True)
+
+    with paths.set_temp_cache(tmpdir):
+        dldir1, urlmapfn1 = _get_download_cache_locs()
+        check_download_cache(check_hashes=True)
+        assert dldir1 != dldir0
+        assert urlmapfn1 != urlmapfn0
+
+    dldir2, urlmapfn2 = _get_download_cache_locs()
+    check_download_cache(check_hashes=True)
+    assert dldir2 != dldir1
+    assert urlmapfn2 != urlmapfn1
+    assert dldir2 == dldir0
+    assert urlmapfn2 == urlmapfn0
```

Comment:
Nitpick: By checking `dldir2 == dldir0` and `urlmapfn2 == urlmapfn0`, doesn't this also indirectly test for `dldir2 != dldir1` and `urlmapfn2 != urlmapfn1` since `dldir1 != dldir0` and `urlmapfn1 != urlmapfn0` is already established by now?

Reply:
A little bit redundant, yes, but I wanted it to be obvious what went wrong - no one should have to apply logic to see which condition has failed.